### ISSUE TITLE
In windows with IPv6 enabled , hostname do not resolve locally as pointing to Link-local-ipv6.

### ISSUE THREAD

**Issue by darshinimashar** [tailscale/tailscale#1682]
Windows machines (some WS 2016/2019 and Windows 10) where we attempt to resolve the hostname of a device from that device will attempt to resolve to the link-local IPv6 address while connected to Tailscale, which does not appear to resolve to anything.

Use case: In Windows Admin Center, where we would get an error that WinRM could not connect to the system using its hostname unless we connect using localhost or its Tailscale IPv4 address - then it connects successfully.

Work Around: Disable IPv6 from the network adapter/removing the link-local address from PowerShell (Remove-NetIPAddress), but it would re-enable upon startup in some cases.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_30c0h)

**Comment by DentonGentry** [comment]
I've installed a new Win10 Professional VM on a network with IPv6 enabled. I cannot run the Admin Center example described, though if need be I think I can install a Windows Server 2019 VM.

So far still looking for a way to produce the problem. The tailscale interface appears in ipconfig and has its usual f77a:115c:a1e0:ab12:: IPv6 ULA.

**Comment by bradfitz** [comment]
The user reported in the email that it resolved to a "link-local address", but fd7a:115c:a1e0:ab12:xxxx:xxxx:xxxx:xxxx is not a link-local address... that's the Tailscale node ULA address.

I don't think we have enough concrete details here to warrant an open bug. We'd need some repro steps first.

**Comment by theh4x0r4chan** [comment]
Hi there! I'm the user with the issue.

I managed to reproduce this issue on two different Tailscale accounts, it was easily reproducible from my end by simply pinging the hostname of the device I was on. The fd7a* address listed under IPv6 Address in ipconfig is indeed resolvable, but that's not what the hostname resolves to if you ping it. In this case if you ping the device's hostname, it resolves to an fe80:99d0* address which is the link-local address for the Wintun adapter - that one does not resolve to anything.

@DentonGentry - this issue for me affected Windows Server 2016/2019 and Windows 10 from my testing so it's not something localized to the Server versions.

Please let me know if I can clarify this issue further.

**Comment by DentonGentry** [comment]
Ah:
<img width="557" alt="Screen Shot 2021-04-09 at 5 00 29 PM" src="https://user-images.githubusercontent.com/1140756/114251221-1f388600-9955-11eb-8902-a53b375087d8.png">

**Comment by DentonGentry** [comment]
If I uninstall Tailscale, ping of the Ethernet's IPv6 link-local address does work.

<img width="542" alt="Screen Shot 2021-04-09 at 5 19 14 PM" src="https://user-images.githubusercontent.com/1140756/114251962-dcc47880-9957-11eb-9d56-b9d91106fc74.png">

**Comment by theh4x0r4chan** [comment]
Thank you for confirming the issue me, Denton. Screenshot attached is the workaround that was mentioned earlier - removing the IPv6 link-local address from the Wintun adapter falls back to the default Ethernet interface's IPv6 address.

![image](https://user-images.githubusercontent.com/44983504/114252641-54a09c00-9974-11eb-9078-8d1642d8f2fb.png)

**Comment by DentonGentry** [comment]
So far as I can tell the name resolution is working as expected here. Windows chooses to resolve the local hostname to the IPv6 link-local address of the first interface. We might wonder if there are other name resolution strategies here, but what Windows does seems legitimate.

I think the problem is that for wintun interfaces, sending to one's own IPv6 link local address is not looping back. When wintun is removed, Ethernet interfaces do loop back their own address.

**Comment by DentonGentry** [comment]
Not fixed in 1.18.
<img width="513" alt="Screen Shot 2021-11-09 at 5 38 49 PM" src="https://user-images.githubusercontent.com/1140756/141033783-f55fe8a7-edc7-4b36-bb43-16d12266eab6.png">

**Comment by DentonGentry** [comment]
Fixed in 1.26 by https://github.com/tailscale/tailscale/commit/2903d42921404777feb3b573487e66a5211dd87d

```
C:\Users\dgentry>ping win10

Pinging win10 [fe80::df2:4aea:4959:52e%9] with 32 bytes of data:
Reply from fe80::df2:4aea:4959:52e%9: time<1ms
Reply from fe80::df2:4aea:4959:52e%9: time<1ms
Reply from fe80::df2:4aea:4959:52e%9: time<1ms

Ping statistics for fe80::df2:4aea:4959:52e%9:
    Packets: Sent = 3, Received = 3, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 0ms, Maximum = 0ms, Average = 0ms
```
